internal/handlers/pg/msg_find.go

```diff
@@ -76,7 +76,7 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 	// Sorting requires fetching all documents and sorting them in memory unless `EnableSortPushdown` is set.
 	// Limit pushdown is not applied when `sort` is set but `EnableSortPushdown` is not set.
 	// Skip pushdown is not supported yet, limit pushdown is not applied when `skip` is non-zero value.
-	if (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
+	if params.Filter.Len() == 0 && (params.Sort.Len() == 0 || h.EnableSortPushdown) && params.Skip == 0 {
```

Comment:
Oh my dear, that's a really serious mistake of mine :disappointed:

Reply:
@chilagrow Please take over this PR tomorrow. At the very least, we should update those comments and add tests. Maybe something else.

Reply:
Yes sure, tests, documentation and comments should be updated.

Do we plan a patch release tomorrow?

Reply:
We will release v1.6.1 today, but that PR should be cleaned-up for v1.7.0